test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/regex/RegexMatcherTest.scala

```diff
@@ -269,43 +269,41 @@ class RegexMatcherTest  {
   @Test def region(): Unit = {
     val matcher0 = Pattern.compile("S[a-z]+").matcher("A Scalable Solution")
 
-    val region0to3 = matcher0.region(0, 3)
-    assertEquals(0, region0to3.regionStart)
-    assertEquals(3, region0to3.regionEnd)
-    assertFalse(region0to3.find())
-
-    val region0to15 = matcher0.region(0, 15)
-    assertEquals(0, region0to15.regionStart)
-    assertEquals(15, region0to15.regionEnd)
-    assertTrue(region0to15.find())
-    assertEquals("Scalable", region0to15.group)
-
-    val region2to7 = region0to15.region(2, 7)
-    assertEquals(2, region2to7.regionStart)
-    assertEquals(7, region2to7.regionEnd)
-    assertTrue(region2to7.find())
-    assertEquals("Scala", region2to7.group)
-
-    val region5toEnd = matcher0.region(5, matcher0.regionEnd)
-    assertEquals(5, region5toEnd.regionStart)
-    if (!executingInJVM) {
-      assertEquals(19, region5toEnd.regionEnd)
-      assertTrue(region5toEnd.find())
-      assertEquals("Solution", region5toEnd.group)
-    }
+    matcher0.region(0, 3)
```

Comment:
We should assert that the returned matcher is indeed the same. At least once:
```suggestion
    assertSame(matcher0, matcher0.region(0, 3))
```